packages/venia-ui/lib/RootComponents/Category/category.js

```diff
@@ -36,14 +55,60 @@ const Category = props => {
 
     const [runQuery, queryResponse] = useLazyQuery(GET_CATEGORY);
     const { loading, error, data } = queryResponse;
+    const { search } = useLocation();
+
+    // Keep track of the search terms so we can tell when they change.
+    const [previousSearch, setPreviousSearch] = useState(search);
+
+    // Get "allowed" filters by intersection of schema and aggregations
+    const { data: introspectionData, error: introspectionError } = useQuery(
+        FilterIntrospectionQuery
+    );
+
+    useEffect(() => {
+        if (introspectionError) {
+            console.error(introspectionError);
+        }
+    }, [introspectionError]);
+
+    // Create a type map we can reference later to ensure we pass valid args
+    // to the graphql query.
+    // For example: { category_id: 'FilterEqualTypeInput', price: 'FilterRangeTypeInput' }
+    const filterTypeMap = useMemo(() => {
+        const typeMap = new Map();
+        if (introspectionData) {
+            introspectionData.__type.inputFields.forEach(({ name, type }) => {
+                typeMap.set(name, type.name);
+            });
+        }
+        return typeMap;
+    }, [introspectionData]);
 
     // Run the category query immediately and whenever its variable values change.
     useEffect(() => {
+        // Wait until we have the type map to fetch product data.
+        if (!filterTypeMap.size) {
+            return;
+        }
+
+        const filters = getFiltersFromSearch(search);
+
+        // Construct the filter arg object.
+        const newFilters = {};
+        filters.forEach((values, key) => {
+            newFilters[key] = getFilterInput(values, filterTypeMap.get(key));
+        });
+
+        // TODO: Category filtering through aggregations seems incorrect. The result from a query using the `in` filter type results in a union of all results in all selected categories. There is no way to get a union of all results AND an intersection with the current category. For now, just search using the current page's category id.
```

Comment:
If we care about category filtering when on a category page we should turn this into a ticket, otherwise I can update this comment to reflect what is going on and why.

cc @awilcoxa @soumya-ashok @schensley

Reply:
Oops! The comment was not meant to remain. Any `TODO` comments I add while working are an indicator for myself to get back to them. In this case I actually turned price and category filtering into two new tickets:

https://jira.corp.magento.com/browse/PWA-403
https://jira.corp.magento.com/browse/PWA-404

I'll replace the comment with a reference.